gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py

```diff
@@ -2030,13 +2039,25 @@ def impl(context, filename, some, output):
         err_str = "xx Expected stdout string '%s' and found: '%s'" % (regexStr, contents)
         raise Exception(err_str)
 
+@given('verify that pg_hba.conf file has "{type}" entries in each segment (primary and mirror) data directories')
+@then('verify that pg_hba.conf file has "{type}" entries in each segment (primary and mirror) data directories')
+def impl(context, type):
+    verify_pg_hba_in_each_segment_data_directories(context, type, False)
+
 @given('verify that pg_hba.conf file has "{type}" entries in each segment data directories')
 @then('verify that pg_hba.conf file has "{type}" entries in each segment data directories')
 def impl(context, type):
+    verify_pg_hba_in_each_segment_data_directories(context, type)
+
+def verify_pg_hba_in_each_segment_data_directories(context, type, check_only_primary=True):
+    if check_only_primary:
+        query = "SELECT hostname, datadir FROM gp_segment_configuration WHERE role='p' AND content > -1;"
+    else:
+        query = "SELECT hostname, datadir FROM gp_segment_configuration WHERE content > -1;"
+
     conn = dbconn.connect(dbconn.DbURL(dbname='template1'), unsetSearchPath=False)
     try:
-        curs = dbconn.query(conn,
-                            "SELECT hostname, datadir FROM gp_segment_configuration WHERE role='p' AND content > -1;")
```

Comment:
Why not just remove `role='p' AND` from this query and use this step in new test?